core/src/main/java/apoc/export/cypher/MultiStatementCypherSubGraphExporter.java

```diff
@@ -388,11 +389,10 @@ private long countArtificialUniques(Iterable<Node> n) {
     private long getArtificialUniques(Node node, long artificialUniques) {
         Iterator<Label> labels = node.getLabels().iterator();
         boolean uniqueFound = false;
-        while (labels.hasNext()) {
+        while (labels.hasNext() && !uniqueFound) {
```

Comment:
Was this bug fix only a speed-up or can we add a test for it?

Reply:
It was both I guess, I added it for the speed, although the bug wouldn't have affected the correctness of the outputted queries. I can add a test for it though :)